pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller_test.go

```diff
@@ -342,6 +372,39 @@ func TestReconcileClusterDeployment(t *testing.T) {
 				r.Delete(gomock.Any()).Times(0)
 			},
 		},
+		{
+			name: "Test Hibernation",
+			localObjects: []runtime.Object{
+				hibernatingClusterDeployment(),
+				testSecret(),
+				testSecretRef(),
+				testSyncSet(),
+				testDeadMansSnitchIntegration(),
+			},
+			expectedSyncSets: &SyncSetEntry{
+				name:                     testClusterName + "-" + snitchNamePostFix + "-" + config.RefSecretPostfix,
+				referencedSecretName:     testClusterName + "-" + snitchNamePostFix + "-" + config.RefSecretPostfix,
+				clusterDeploymentRefName: testClusterName,
+			},
+			expectedSecret: &SecretEntry{
+				name:                     testClusterName + "-" + snitchNamePostFix + "-" + config.RefSecretPostfix,
+				snitchURL:                testSnitchURL,
+				clusterDeploymentRefName: testClusterName,
+			},
+			verifySyncSets: verifyNoSyncSet,
+			verifySecret:   verifyNoSecret,
+			setupDMSMock: func(r *mockdms.MockClientMockRecorder) {
+				r.FindSnitchesByName(gomock.Any()).Return([]dmsclient.Snitch{
+					{Token: testSnitchToken},
+				}, nil)
+				r.Delete(gomock.Any()).Return(true, nil)
```

Comment:
```suggestion
				r.FindSnitchesByName(gomock.Any()).Return([]dmsclient.Snitch{
					{Token: testSnitchToken},
				}, nil).Times(1)
				r.Delete(gomock.Any()).Return(true, nil).Times(2)
```

I might have the counts wrong, but you need to verify the mocks are actually called an expected number of times.  We expect to find the snitch once before delete.  And expect to issue a delete on the syncset and secret.

Reply:
`Times(1)` is implied, so there is no need to check in that case. I have added the checks back in for consistency with the rest of the tests though - https://github.com/openshift/deadmanssnitch-operator/pull/117/commits/fddfa6c29c5263260d2d365bcccdead8cca7176e (This is just the dms client mock though so we only fetch once and delete once.)


The secret and syncset deletion is verified with https://github.com/openshift/deadmanssnitch-operator/blob/fddfa6c29c5263260d2d365bcccdead8cca7176e/pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller_test.go#L394-L395